qupath-core/src/main/java/qupath/lib/objects/utils/MeasurementStrategy.java

```diff
@@ -0,0 +1,144 @@
+            );
+        }
+    });
+
+    /**
+     * Calculate a mean weighted by object area and pass this to the output object.
+     */
+    MeasurementStrategy WEIGHTED_MEAN = ((pathObjects, outputList) -> {
+        var weights = computeAreaWeights(pathObjects);
+        for (var name: pathObjects.getFirst().getMeasurementList().getNames()) {
+            outputList.put(name,
+                    pathObjects.stream()
+                            .mapToDouble(po ->
+                                    po.getMeasurementList().get(name) * weights.get(po)
+                            )
+                            .sum()
+            );
+        }
+    });
+
+    private static Map<PathObject, Double> computeAreaWeights(List<? extends PathObject> pathObjects) {
+        Logger logger = LoggerFactory.getLogger(MeasurementStrategy.class);
+        Map<PathObject, Double> output = new HashMap<>();
+        double sum = 0;
+        for (var po: pathObjects) {
+            double area = po.getROI().getArea();
+            sum += area;
+            output.put(po, area);
+        }
+        if (sum == 0) {
+            logger.warn("Zero areas");
+        }
+        for (var key: output.keySet()) {
+            double finalSum = sum;
+            output.computeIfPresent(key, (k, value) -> value / finalSum);
+        }
+        return output;
+    }
+
+    /**
+     * Calculate the median of each measurement and pass these to the output object.
+     */
+    MeasurementStrategy MEDIAN = ((pathObjects, outputList) -> {
+        for (var name: pathObjects.getFirst().getMeasurementList().getNames()) {
+            outputList.put(
+                    name,
+                    pathObjects.stream()
+                            .mapToDouble(po -> po.getMeasurementList().get(name))
+                            .sorted()
+                            .skip((pathObjects.size() - 1) / 2)
+                            .limit(2 - pathObjects.size() % 2)
+                            .average()
+                            .orElse(Double.NaN)
+            );
+        }
+    });
+
+    /**
+     * Choose an object at random and use the measurements from this object.
+     */
+    MeasurementStrategy RANDOM = ((pathObjects, outputList) -> {
+        var chosen = pathObjects.stream()
+                .skip((int)(pathObjects.size() * Math.random()))
+                .findFirst()
+                .map(PathObject::getMeasurementList)
+                .orElse(null);
+        if (chosen == null) {
+            return;
+        }
+        outputList.putAll(chosen);
+    });
+
+    /**
+     * Use the measurements of the largest object by area, length or number of points.
+     */
+    MeasurementStrategy USE_BIGGEST = ((pathObjects, outputList) -> {
+        var chosen = pathObjects.stream()
+                .max(
+                        Comparator.comparingDouble(
+                        pathObject -> {
+                            if (pathObject.getROI().getArea() > 0) {
```

Comment:
This looks wrong to me, I think it should be something like
```java
Comparator.comparingDouble(p -> p.getROI().getArea())
    .thenComparing(p -> p.getROI().getLength())
    .thenComparing(p -> p.getROI().getNumPoints())
```
Otherwise we risk comparing lengths with areas with numbers of points, rather than preferring non-zero areas, then lengths, the point counts..